### DISCUSSION TITLE
Unable to run version 0.6.0 of litiengine, problem with JAVA 21

### DISCUSSION THREAD

**Discussion by bkebir** [gurkenlabs/litiengine#796] [General]
**Describe the bug**
When you run the project that you haven't been able to run for 3 weeks (since you updated the project)

**Stack Trace**
The error message you get either in your console or the `crash.txt` located in your program's root directory.
```Program.java is a non-project file, only syntax errors are reported```

**To Reproduce**
1. Go to 'run'

**Expected behavior**
I want to be able to launch the project.

**Screenshots**
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/d1d850de-32a2-4ae3-986f-143507b1cd3f)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/fcc7b991-4051-4963-9719-f9b4db46f44c)

**Your System:**
 - OS: [e.g. Windows]
 - LITIENGINE version: [e.g. 0.6.0]
 - Java JDK/JRE version: [e.g. 21.0.1 2023-10-17] 
 - Screen resolution: [e.g. 1920x1080]

**Additional context**
I've updated my java to version 21 but it still doesn't work, but it works again when you go back to an earlier version of the project

**Comment by bkebir** [comment]
We think that the problem stems from a dependency to which we no longer have access, which may have been public at the outset and has now been made private ``de.gurkenlabs.semverguru:de.gurkenlabs.semverguru.gradle.plugin:1.3\r\nNo matching variant of de.gurkenlabs:semverguru:1.3 was found.``

**Comment by nightm4re94** [comment]
This is [not the case](https://plugins.gradle.org/plugin/de.gurkenlabs.semverguru).
The error message you've mentioned tells me that you are not compiling with Java 21. Please make sure that not only the project JDK is Java 21, but also that Gradle is using the project JDK.

**Comment by nightm4re94** [comment]
Have you been able to resolve this?

**Comment by Maxime91680** [comment]
Unfortunately not, we have the right JDK but we don't know if gradle can retrieve the right JDK, we've also tried adding dependencies and plugins as you advised but nothing works

**Comment by nightm4re94** [comment]
Is there a reason you are preferring VSCode over  a full Java IDE such as IntelliJ or Eclipse?
The Gradle integration in IntelliJ, for example, is much better and the community can be more helpful if you're using a similar toolchain.

**Comment by Hades948** [comment]
I was able to successfully upgrade my project to Java 21 and I use VSCode. I'd be willing to try to help here, but I need more information about how you're running your program. Are you just pressing F5 and letting VS handle the launch? Or are you using a makefile or something? It'd be helpful to see the entire console output when the program is launched (inluding the command used for the launch if VS is driving it).

That said, VSCode with Java is a frustrating headache. I deal with it, I guess, because I'm stubborn. But if you want to save yourself a headache, you should look into Eclipse or IntelliJ like @nightm4re94 suggested.

**Reply by iamllcoolray** [reply]
I'm using IntelliJ; we are using different development tools; how did you upgrade the project to Java 21? I'm looking for insight to see if I am missing something. 🧐

**Reply by Hades948** [reply]
Like you said, I'm not using IntelliJ. Unfortunately, I'm not sure how to set the java version in IntelliJ. I'd recommend asking in the Discord. I imagine someone in there might know.

**Comment by bkebir** [comment]
Hi @Hades948,

Sorry for the late reply, here's our problem. When we launch the project using the F5 key or by pressing the "RUN" button, we get an error telling us "BUILD FAILED" and this only applies since we updated our java to version 21.

![image](https://github.com/gurkenlabs/litiengine/assets/64213378/805a705c-c642-4f92-b20d-edc0abdb5c52)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/6ae747ad-9221-4371-a10a-bc0fc3c4cf36)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/29a55e44-a3dc-435b-80ec-9c1cc995c543)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/af3ae6a1-f977-41f1-83da-8e2cabfbd34d)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/66879b40-69d2-479a-938f-00f2bfe6b4cc)
![image](https://github.com/gurkenlabs/litiengine/assets/64213378/b8daf269-d0c0-4e7c-9f48-64fc3a399ea1)

**Reply by Hades948** [reply]
Hmm, unfortunately, I am also using Maven instead of Gradle (as I said, I'm stubborn). This looks pretty specific to Gradle, so I'm not sure if I'll be able to help here. At least as far as I'm aware, it doesn't seem like there's anyone active in the community using VS Code with Gradle. So, to get the most help, I'd recommend switching to VSCode + Maven or IntelliJ/Eclipse + Gradle. I would strongly **strongly** suggest the latter option since that's what the creators of the engine use so you'll be able to get more support using the platforms suggested by nightm4re94. And I wouldn't recommend being as stubborn as I am :)

**Comment by nightm4re94** [comment]
I'm sorry, but I am not able to reproduce your problem :/

**Reply by nightm4re94** [reply]
Found the issue!
The SemVerGuru plugin currently has a bug that prevents it from being applied to any repository without git tags.
Make sure you also check out tags when cloning LITIENGINE to work around this issue. 
I have [opened a ticket](https://codeberg.org/nightm4re/SemVerGuru/issues/1) in the plugin repo for this.

**Reply by iamllcoolray** [reply]
Really?!?!!?? Thank you so much! I was able to get it working. What I ended up doing was cloning from the parent repository.

**Reply by bkebir** [reply]
Hi @iamllcoolray 
Could you tell us the steps to take to clone from the parent repository. Thank you very much.

**Reply by bkebir** [reply]
We are on visual studio for reminder

**Reply by iamllcoolray** [reply]
Sure. I first deleted the cloned fork. and then did the following below.

```
> git clone https://github.com/gurkenlabs/litiengine.git

> git remote rm origin

> git remote add origin https://github.com/[username]/litiengine.git
```

I fired up Intellij and Gradle was able to successfully build the project on both my Windows and Mac machines.

Honestly wish I thought about doing that sooner. 🤣

I hope this helps. 

*** Disclaimer: I am running Intellij 2023.3.2 ***

**Reply by nightm4re94** [reply]
The issue was fixed in the SemVerGuru plugin now, I've released version 1.4.
